compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -218,7 +218,12 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
   private def makeParamsString(sym: Symbol, reflProxy: Boolean,
       inRTClass: Boolean): String = {
     val tpe = sym.tpe
-    val paramTypeNames = tpe.params map (p => internalName(p.tpe))
+
+    val paramTypeNames0 = tpe.params map (p => internalName(p.tpe))
+    val paramTypeNames =
+      if (!isScalaJSDefinedJSClass(sym.owner)) paramTypeNames0
+      else internalName(sym.owner.toTypeConstructor) :: paramTypeNames0
+
```

Comment:
Hum, good catch. Generalizing `inRTClass` seems risky, but I can definitely factor out the ways the first parameter type is prepended to the signature.